### PR TITLE
DSRE-384: Fix capitalization in advertisers

### PR DESCRIPTION
**Pull request by rebecca-burwei**
I'm seeing inconsistent capitalization in the advertiser field.

Checklist for reviewer:

- [ ] Commits should reference a bug or github issue, if relevant (if a bug is referenced, the pull request should include the bug number in the title)
- [ ] If the PR comes from a fork, trigger integration CI tests by running the [Push to upstream workflow](https://github.com/mozilla/bigquery-etl/actions/workflows/push-to-upstream.yml) and provide the `<username>:<branch>` of the fork as parameter. The parameter will also show up
in the logs of the `manual-trigger-required-for-fork` CI task together with more detailed instructions.
- [ ] If adding a new field to a query, ensure that the schema and dependent downstream schemas have been updated

For modifications to schemas in restricted namespaces (see [`CODEOWNERS`](/CODEOWNERS)):
- [ ] Follow the [change control procedure](https://docs.google.com/document/d/1TTJi4ht7NuzX6BPG_KTr6omaZg70cEpxe9xlpfnHj9k/edit#heading=h.ttegrcfy18ck)

### CODE REVIEW

**Comment by whd** [comment]
I believe this PR is subject to the [change control procedure](https://docs.google.com/document/d/1TTJi4ht7NuzX6BPG_KTr6omaZg70cEpxe9xlpfnHj9k/edit#heading=h.ttegrcfy18ck) for Firefox Suggest data that is mentioned at the end of the PR template, because it changes a view into a source table that is accessible to`workgroup:search-terms/sanitized` (`moz-fx-data-shared-prod.search_terms_derived.suggest_impression_sanitized_v1`), and that view is accessible to `workgroup:contextual-services` (`moz-fx-data-shared-prod.contextual_services.quicksuggest_impression`).

This will be the first case where we use the change control procedure. Please follow the document's instructions and file a DSRE Jira issue, though note that per https://mozilla-hub.atlassian.net/browse/DSRE-344?focusedCommentId=495411 the flow is still being worked on and if you run into issues with the template we can sort them out.

File: sql/moz-fx-data-shared-prod/contextual_services/event_aggregates/view.sql [hunk 1]
```diff
@@ -2,6 +2,7 @@ CREATE OR REPLACE VIEW
   `moz-fx-data-shared-prod.contextual_services.event_aggregates`
 AS
 SELECT
-  *
+  * EXCEPT (advertiser),
+  LOWER(advertiser) as advertiser
```
Comment (jklukas) [inline sql/moz-fx-data-shared-prod/contextual_services/event_aggregates/view.sql:6]:
```suggestion
  * REPLACE(LOWER(advertiser) AS advertiser)
```

The REPLACE construction is a bit more elegant for this case where the field name is staying the same.

File: sql/moz-fx-data-shared-prod/contextual_services/quicksuggest_impression/view.sql [hunk 2]
```diff
@@ -2,6 +2,7 @@ CREATE OR REPLACE VIEW
   `moz-fx-data-shared-prod.contextual_services.quicksuggest_impression`
 AS
 SELECT
-  * EXCEPT (search_query, matched_keywords) REPLACE(mozfun.norm.metadata(metadata) AS metadata)
+  * EXCEPT (search_query, matched_keywords, advertiser) REPLACE(mozfun.norm.metadata(metadata) AS metadata),
+  LOWER(advertiser) as advertiser
```
Comment (jklukas) [inline sql/moz-fx-data-shared-prod/contextual_services/quicksuggest_impression/view.sql:6]:
```suggestion
  * EXCEPT (search_query, matched_keywords) REPLACE(mozfun.norm.metadata(metadata) AS metadata, LOWER(advertiser) as advertiser)
```

**Review by jklukas — APPROVED** [review]
Just two style suggestions.

File: sql/moz-fx-data-shared-prod/contextual_services/event_aggregates/view.sql (hunk 1 above)
Comment (rebecca-burwei) [reply sql/moz-fx-data-shared-prod/contextual_services/event_aggregates/view.sql:6]: @jklukas thanks! I saw REPLACE being used elsewhere, but when I looked it up in the bigquery documentation, I only found this [string function](https://cloud.google.com/bigquery/docs/reference/standard-sql/string_functions#replace) -- it made me wonder: is a different flavor of SQL being used here?

**Comment by rebecca-burwei** [comment]
@whd thanks, I've opened a ticket here: https://mozilla-hub.atlassian.net/browse/DSRE-384

File: sql/moz-fx-data-shared-prod/contextual_services/event_aggregates/view.sql (hunk 1 above)
Comment (jklukas) [reply sql/moz-fx-data-shared-prod/contextual_services/event_aggregates/view.sql]: REPLACE is overloaded, which can be confusing. The `* REPLACE` syntax is documented under SELECT: https://cloud.google.com/bigquery/docs/reference/standard-sql/query-syntax#modifiers_for_operator

**Comment by jklukas** [comment]
Note that @ncloudioj just filed https://bugzilla.mozilla.org/show_bug.cgi?id=1748848 which should make client values consistent. So I believe the changes here will eventually be a no-op, but this is still valuable for cleaning the data short-term.

**Comment by rebecca-burwei** [comment]
> Note that @ncloudioj just filed https://bugzilla.mozilla.org/show_bug.cgi?id=1748848 which should make client values consistent. So I believe the changes here will eventually be a no-op, but this is still valuable for cleaning the data short-term.

thanks @jklukas . looks like that bug is closed/fixed, but I'm still seeing mixed capitalization in quicksuggest_impression.

**Comment by ncloudioj** [comment]
> looks like that bug is closed/fixed, but I'm still seeing mixed capitalization in quicksuggest_impression.

FWIW, that bugfix landed in 97 Nightly (now in beta). Fx 96 (released today) doesn't have the fix.

Is it possible to use some workarounds for now?

**Comment by rebecca-burwei** [comment]
> Is it possible to use some workarounds for now?

Yep, we need this by next week, so we'll use a workaround

**Comment by jklukas** [comment]
I just ran `bqetl format sql` on this and pushed the results so that we can pass CI.

@mreid-moz has approved in https://mozilla-hub.atlassian.net/browse/DSRE-384 ; next step is for SRE to review and merge.

**Comment by rebecca-burwei** [comment]
thanks @jklukas - I don't know if other files need to be updated so that this change runs on data we've already collected as well as new incoming data

**Review by whd — APPROVED** [review]
`quicksuggest_impression_live` isn't included here so there is a difference of behavior between the live and stable views; if that's expected I can proceed with deploying this, otherwise it should be added.

**Comment by jklukas** [comment]
The failure of the `integration` CI step looks spurious. It should be able to succeed on retry.

**Pull request merged**